tracer/src/Datadog.Trace/DatabaseMonitoring/DatabaseMonitoringPropagator.cs

```diff
@@ -113,21 +113,33 @@ internal static string PropagateDataViaComment(DbmPropagationLevel propagationSt
         /// Currently only working for MSSQL (uses an instruction that is specific to it)
         /// </summary>
         /// <returns>True if the traceparent information was set</returns>
-        internal static bool PropagateDataViaContext(DbmPropagationLevel propagationLevel, IntegrationId integrationId, IDbConnection? connection, Span span)
+        internal static bool PropagateDataViaContext(DbmPropagationLevel propagationLevel, IntegrationId integrationId, IDbCommand command, Span span)
         {
-            if (propagationLevel != DbmPropagationLevel.Full || integrationId != IntegrationId.SqlClient || connection == null)
+            if (propagationLevel != DbmPropagationLevel.Full || integrationId != IntegrationId.SqlClient || command.Connection == null)
```

Comment:
Doesn't invoking command.Connection throw an exception in some cases? I believe you've guarded that by putting it _after_ the SqlClient check (for example) I would recommend leaving a comment to that effect too, in case someone refactors 😄

Reply:
I've done that in this PR https://github.com/DataDog/dd-trace-dotnet/pull/6210/files

But maybe what I should do is just merge these two PRs together